src/deepqmc/fit.py

```diff
@@ -147,6 +147,7 @@ def fit_wf(  # noqa: C901
                 keep_graph=require_psi_gradient,
             )
             log_ws = 2 * log_psis.detach() - 2 * log_psi0s
+            Es_loc = Es_loc.where(~torch.isinf(log_ws), Es_loc.new_tensor(0))
```

Comment:
Why zero? Shouldn't it be taken out of the sample entirely?

Reply:
The zero importance weight `log_ws = inf` ensure that the value of the local energy is irrelevant, hence we can safely set it to zero without any consequence. Removing the sample entirely from the batch would work in principle, but gives batches of varying size. This causes problems when assigning local energies to the walkers etc. , that would require additional code that keeps track of the replaced samples.

Reply:
Ah, ok, got it. Can we be absolutely certain that `log_ws` is inf whenever `Es_loc` is inf? And if so, could you add a comment about it, so this logic is transparent?

Reply:
I will add a comment. My idea was that this line only removes the local energies that are conceptually save to remove. These are only the ones which have zero importance weights. In my experiments there where no other NaNs in the local energies except for the samples with zero weight. If however NaNs appear due to another reason and their sample weight wasn't zero we would want to see them comming up and solve this elsewhere. This ensures that we do not introduce a bias in the energy.

Reply:
I'm sorry, I just can't read 🤦🏻‍♂️ I was thinking you mask by whether the energies are inf... This makes of course perfect sense as is, not need to add a comment